pymatgen/io/openmm/utils.py

```diff
@@ -360,111 +360,182 @@ def parameterize_system(
-            "smirnoff99Frosst-1.1.0",
-            "smirnoff99Frosst-1.0.4",
-            "smirnoff99Frosst-1.0.8",
-            "smirnoff99Frosst-1.0.6",
-            "smirnoff99Frosst-1.0.3",
-            "smirnoff99Frosst-1.0.1",
-            "smirnoff99Frosst-1.0.5",
-            "smirnoff99Frosst-1.0.9",
-            "smirnoff99Frosst-1.0.7",
-            "openff-1.0.1",
-            "openff-1.1.1",
-            "openff-1.0.0-RC1",
-            "openff-1.2.0",
-            "openff-1.3.0",
-            "openff-2.0.0-rc.2",
-            "openff-2.0.0",
-            "openff-1.1.0",
-            "openff-1.0.0",
-            "openff-1.0.0-RC2",
-            "openff-1.3.1",
-            "openff-1.2.1",
-            "openff-1.3.1-alpha.1",
-            "openff-2.0.0-rc.1",
-            "gaff-1.4",
-            "gaff-1.8",
-            "gaff-1.81",
-            "gaff-2.1",
-            "gaff-2.11",
-        ]
         small_molecules = {}
         large_or_water = {}
+        # iterate through each smile, if no forcefielded provided use Sage
         # iterate through each molecule and forcefield input as list
-        for smile, ff_name in force_field.items():
+        # Add charges to the molecule if provided
+        for smile in smile_strings:
             openff_mol = openff.toolkit.topology.Molecule.from_smiles(smile)
-            # Assign mols and forcefield as small molecule vs AMBER or
-            # CHARMM
-            if ff_name.lower() in small_ffs:
-                small_molecules[openff_mol] = ff_name.lower()
+            found_isomorphic = False
+            if len(partial_charges) > 0:
+                for mol, charges in partial_charges:
+                    inferred_mol = infer_openff_mol(mol)
+                    is_isomorphic, atom_map = get_atom_map(inferred_mol, openff_mol)
+                    if is_isomorphic:
+                        charged_mol = assign_charges_to_mols(
+                            smile_strings=[smile],
+                            partial_charges=[(mol, charges)],
+                            partial_charge_method=partial_charge_method,
+                            partial_charge_scaling=partial_charge_scaling,
+                        )[0]
+                        found_isomorphic = True
+                if found_isomorphic:
+                    mol_to_load = charged_mol
+                else:
+                    mol_to_load = openff_mol
             else:
-                large_or_water[openff_mol] = ff_name.lower()
-        forcefield_omm = omm_ForceField()
+                mol_to_load = openff_mol
+            # Assign ff to each molecule
+            if smile in force_field.keys():
+                ff_name = force_field[smile]
+                if ff_name.lower() in all_small_ffs or ff_name.lower() in basic_small_ffs:
+                    small_molecules[mol_to_load] = ff_name.lower()
+                else:
+                    large_or_water[mol_to_load] = ff_name.lower()
```

Comment:
I actually think that "large" isn't really a category here. My understanding is that "small molecule" force fields use "small" to distinguish them from coarse grained or protein force fields, since we aren't supporting those, I think that our molecules are either "small molecules" or water. I think that if we reorganized this a bit to treat water as one case and not-water as another, it might be a bit cleaner.

Reply:
I'm using "large" to allow for Amber and CHARMM DNA and protein models.

Reply:
Got it, thanks for clarifying. In that case, ignore my comment. Hopefully someone will use this for large molecules lol.

On that note "large molecules" doesn't feel like the right term. What would be more appropriate? Biomolecules? Should we change it in the code?

Reply:
Yeah I wish I knew the field better. Based on the OpenMM documentation, 

> "The OpenMM force fields above include pregenerated templates for biopolymers and solvents. If your system instead contain small molecules, it is often necessary to generate these parameters on the fly."

So may "biopolymers_and_solvents" might be best

Reply:
I like what you settled on here.